rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCostAnnotator.scala

```diff
@@ -0,0 +1,193 @@
+    val keyStore = KeyStore.getInstance("PKCS12")
+    keyStore.load(null)
+    factory.init(keyStore, Array.empty)
+    val km: Array[KeyManager] = factory.getKeyManagers
+    val tm: Array[TrustManager] =
+      PemManagersProvider.buildTrustManagers(certificates)
+    val random: SecureRandom = new SecureRandom
+    val sslContext = SSLContext.getInstance("TLSv1.2")
+    sslContext.init(km, tm, random)
+    sslContext
+  }
+
+  implicit val dispatcher: ExecutionContextExecutor = context.system.dispatcher
+  def receive = onMessage(0, SortedSet.empty(Member.ageOrdering))
+
+  private def onMessage(deletionCost: Int, membersByAgeDesc: SortedSet[Member]): Receive = {
+    case cs @ ClusterEvent.CurrentClusterState(members, _, _, _, _) =>
+      log.debug("Received CurrentClusterState {}", cs)
+      updateIfNewCost(deletionCost, membersByAgeDesc ++ members) // ordering used is from the first operand (so, by age)
+
+    case ClusterEvent.MemberUp(m) =>
+      log.debug("Received MemberUp {}", m)
+      updateIfNewCost(deletionCost, membersByAgeDesc + m)
+
+    case ClusterEvent.MemberRemoved(m, _) =>
+      log.debug("Received MemberRemoved {}", m)
+      updateIfNewCost(deletionCost, membersByAgeDesc - m)
+
+    case PodAnnotated() =>
+      log.debug("Annotation updated successfully to {}", deletionCost)
+      // cancelling an eventual retry in case the annotation succeeded in the meantime
+      timers.cancel(RetryTimerId)
+
+    case ScheduleRetry(ex) =>
+      log.error(s"Failed to update annotation: [$ex]. Retrying with fixed delay of {}.", costSettings.retryDelay)
+      timers.startSingleTimer(RetryTimerId, RetryAnnotate, costSettings.retryDelay)
+
+    case GiveUp(er: String) =>
+      log.error(
+        "There was a client error when trying to set pod-deletion-cost annotation. " +
+        "Not retrying, check configuration. Error: {}",
+        er)
+
+    case RetryAnnotate =>
+      costRequestToResult(deletionCost).pipeTo(self)
+
+    case es => log.debug("Ignoring message {}", es)
+  }
+
+  private def costRequestToResult(newCost: Int): Future[RequestResult] = {
+    log.debug(
+      "Updating pod-deletion-cost annotation for pod: [{}] with cost: [{}]. Namespace: [{}]",
+      settings.podName,
+      newCost,
+      podNamespace
+    )
+    val request = ApiRequests.podDeletionCost(settings, apiToken, podNamespace, newCost)
+    if (settings.secure)
+      http.singleRequest(request, clientSslContext)
+    else
+      http.singleRequest(request)
+  }.map {
+      case HttpResponse(status, _, e, _) if status.isSuccess() =>
+        e.discardBytes()(Materializer(context)); PodAnnotated()
+      case HttpResponse(s @ ClientError(_), _, _, _) => GiveUp(s.toString())
+      case HttpResponse(status, _, _, _)             => ScheduleRetry(s"Request failed with status=$status")
+      case _                                         => GiveUp("Unknown")
```

Comment:
Is this even reachable? The above case should capture all. 

I think you should make sure to discard bodies for the non success status cases as well, in case k8 emits some error details (then we'd be clogging up that connection pool).